### ISSUE TITLE
`@angular/animations` may interfer with OverlayRef from `@angular/cdk/overlay`, thus removing DOM nodes

### ISSUE THREAD

**Issue by dherges** [angular/components#7411]
#### Bug, feature request, or proposal:

Proposal / bug

#### What is the expected behavior?

Less confusion :smiley_cat: 

#### What is the current behavior?

Created a overlay with:

```ts
import { Overlay, OverlayRef } from '@angular/cdk/overlay';

class Foo {

  private overlay: OverlayRef;

  constructor(
    private overlay: Overlay
  ) {}

  // this is executed first, so we have a Overlay on screen with `FooComponent` visible
  createOverlayWithComponent() {
    this.overlayRef = overlay.create();
    this.overlayRef.attach(new ComponentPortal<FooComponent>(FooComponent, null, this.injector));
  }

  // Now, w/o calling `this.overlay.dispose()` the DOM nodes of `FooComponent` hostView are removed
  whatCausesDomNodesToBeRemoved() {
    overlayRef.detach();
  }

}

```

![2017-09-29 11_45_33-developer tools - http___localhost_4200_](https://user-images.githubusercontent.com/901354/31010655-e0021104-a50b-11e7-84ed-07548c6fd9b0.png)

The call to `overlayRef.detach()` ends up in an `onRemovalComplete()` callback of `@angular/animations`, thus removing nodes from the DOM. This is the callback being invoked:

https://github.com/angular/angular/blob/master/packages/platform-browser/animations/src/animation_renderer.ts#L31

#### What are the steps to reproduce?

Hard to repro atm. 
Here is a stack trace walk through....

`overlayRef.detach()` calls through to `AnimationEngine.onRemove()` ... then, this marks nodes as "collectedLeaveElements"  .... 

![repro1](https://user-images.githubusercontent.com/901354/31012204-2fc04e04-a511-11e7-9a6f-70ebced2a19e.gif)

And then, in the next application `tick()`, the DOM nodes get removed ... here, I skip from the `AnimationEngine.onRemove()` which was called from `OverlayRef.detach()` to a breakpoint on node removal ... 

![repro2](https://user-images.githubusercontent.com/901354/31012073-c0e8ddfc-a510-11e7-8116-2ca8319956b4.gif)

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?

"@angular/cdk@2.0.0-beta.11":
  version "2.0.0-beta.11"

"@angular/animations@~4.4.3":
  version "4.4.3"

web-animations-js@~2.3.1:
  version "2.3.1"

#### Is there anything else we should know?

Is it intended behaviour? I don't think so!

Is it Material bug?

Is it Animations bug?

**Comment by dherges** [comment]
Here are little bit description what my use case was:

Created an overlay and grab a `OverlayRef`. Then dismiss the overlay and expect DOM nodes to be removed. Thus called `overlayRef.detach()` and (because of above), DOM nodes were removed in the process (most of the time). Some times, the DOM nodes were not removed (this is why issue is hard to repro...). The solution is (obviously) to use `overlayRef.detach()` for detaching Angular data structures from DOM and then `overlayRef.dispose()` to remove the DOM nodes.

**Comment by jelbourn** [comment]
Closing since I think this is obsolete now. Feel free to open a new issue if a problem persists.
